solr/core/src/java/org/apache/solr/response/transform/TransformerFactory.java

```diff
@@ -105,18 +107,17 @@ default boolean mayModifyValue() {
     }
   }
 
-  public static final Map<String, TransformerFactory> defaultFactories = new HashMap<>(9, 1.0f);
+  // loaded once, based on the node
+  private static Map<String, TransformerFactory> builtIns;
 
-  static {
-    defaultFactories.put("explain", new ExplainAugmenterFactory());
-    defaultFactories.put("value", new ValueAugmenterFactory());
-    defaultFactories.put("docid", new DocIdAugmenterFactory());
-    defaultFactories.put("shard", new ShardAugmenterFactory());
-    defaultFactories.put("child", new ChildDocTransformerFactory());
-    defaultFactories.put("subquery", new SubQueryAugmenterFactory());
-    defaultFactories.put("json", new RawValueTransformerFactory("json"));
-    defaultFactories.put("xml", new RawValueTransformerFactory("xml"));
-    defaultFactories.put("geo", new GeoTransformerFactory());
-    defaultFactories.put("core", new CoreAugmenterFactory());
+  /** (for internal use) */
+  public static synchronized Map<String, TransformerFactory> builtIns(NodeConfig nodeConfig) {
```

Comment:
Since the lazy loaded list is initialized based on NodeConfig (basically CoreContainer), admittedly there's a test stability risk since NodeConfig ClassLoader classpath of the first test that runs will "win".  For the built-in plugins, let's just not do anything that would disturb that list.  If we wanted to test that someone could add a plugin in a JAR, that would be a high level integration test using Docker or BATS.  Not unit tests.

Reply:
Anyway, the use of NodeConfig here thus adds a capability of config-less addition of a plugin (i.e. just add a JAR for a custom DocTransformer; no need to edit solrconfig.xml / configSet).